javascript_nodejs/4.simple-prompt-bot/dialogs/mainDialog/index.js

```diff
@@ -61,16 +66,17 @@ class MainDialog {
         // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
         if (context.activity.type === 'message') {
             // Create dialog context
-            const state = this.conversationState.get(context);
-            const dc = this.dialogs.createContext(context, state);
+            const dc = await this.dialogs.createContext(context);
 
             const utterance = (context.activity.text || '').trim().toLowerCase();
             if (utterance === 'cancel') { 
                 await dc.endAll(); 
             }
             
             // Continue the current dialog
-            await dc.continue();
+            if (!context.responded) {
```

Comment:
Without the message being sent to the user indicating that the dialogStack has been answered, this if-block is unnecessary. 

What was the scenario beforehand for adding the if-block? It doesn't look like any other potential messages are being sent...

Reply:
I only just realized that it was wrapped in the condition BECAUSE of the possibility it had just been canceled.  Fixed!